cmd/flightctl-agent/main.go

```diff
@@ -2,27 +2,53 @@ package main
 
 import (
 	"context"
+	"encoding/json"
 	"flag"
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
+	"time"
 
 	"github.com/flightctl/flightctl/internal/agent"
+	"github.com/flightctl/flightctl/internal/agent/device/fileio"
+	"github.com/flightctl/flightctl/pkg/executer"
 	"github.com/flightctl/flightctl/pkg/log"
 	"github.com/flightctl/flightctl/pkg/version"
 )
 
 func main() {
-	if len(os.Args) > 1 && os.Args[1] == "version" {
-		versionInfo := version.Get()
-		fmt.Printf("Flightctl Agent Version: %s\n", versionInfo.String())
-		fmt.Printf("Git Commit: %s\n", versionInfo.GitCommit)
-		os.Exit(0)
+	args := os.Args[1:]
+
+	if len(args) == 0 || isAgentFlags(args) {
+		if hasHelpFlag(args) {
+			printAgentHelp()
+			os.Exit(0)
+		}
+
+		flag.CommandLine = flag.NewFlagSet("agent", flag.ExitOnError)
+		command := NewAgentCommand()
+		if err := command.Execute(); err != nil {
+			os.Exit(1)
+		}
+		return
 	}
 
-	command := NewAgentCommand()
-	if err := command.Execute(); err != nil {
-		os.Exit(1)
+	switch args[0] {
+	case "-h", "--help":
+		printUsage()
+		os.Exit(0)
+
+	case "version":
+		printVersion()
+		os.Exit(0)
+
+	case "system-info":
```

Comment:
note: This command was added to make manual testing easierand could have user value.